### PR TITLE
feat: introduce phpstan php static analysis tool

### PR DESCRIPTION
**Pull request by gbprod**
This introduce builtin diagnostic for [phpstan](https://github.com/phpstan/phpstan), a php static analyser.

### CODE REVIEW

File: lua/null-ls/builtins/diagnostics.lua [hunk 1]
```diff
@@ -436,4 +436,38 @@ M.codespell = h.make_builtin({
...
+        check_exit_code = function(code)
+            return code <= 1
+        end,
+        on_output = function(params)
+            local parser = h.diagnostics.from_json({})
+
+            params.messages = {}
+            if params.output and params.output.files then
+                -- The key is the temp filename, that's why we use for tu get the first element
+                for _, errors in pairs(params.output.files) do
```
Comment (jose-elias-alvarez) [inline lua/null-ls/builtins/diagnostics.lua:458]: This is sensible, but I think it makes sense to add the temp file's path to `params`, which would let you access the necessary nested value directly. Would that fix the issue?

**Review by jose-elias-alvarez — COMMENTED** [review]
Thanks for working on this! Adding the temp file's path to `params` is a trivial change, so let me know if that would solve the issue here.

File: lua/null-ls/builtins/diagnostics.lua [hunk 2]
```diff
@@ -436,4 +436,38 @@ M.codespell = h.make_builtin({
...
 
+M.phpstan = h.make_builtin({
+    method = DIAGNOSTICS,
+    filetypes = { "php" },
+    generator_opts = {
+        command = "./vendor/bin/phpstan",
+        args = { "analyze", "--error-format", "json", "--no-progress", "$FILENAME" },
+        to_stdin = true,
+        to_stderr = false,
+        to_temp_file = true,
```
[the remark this answers is not in the document]
Comment (gbprod) [reply lua/null-ls/builtins/diagnostics.lua:447]: :+1:

File: lua/null-ls/builtins/diagnostics.lua [hunk 3]
```diff
@@ -436,4 +436,38 @@ M.codespell = h.make_builtin({
...
+                -- The key is the temp filename, that's why we use for tu get the first element
+                for _, errors in pairs(params.output.files) do
+                    params.messages = errors.messages
+                    break
+                end
+            end
+
+            return parser({ output = params.messages })
+        end,
+        condition = function(utils)
```
[the remark this answers is not in the document]
Comment (gbprod) [reply lua/null-ls/builtins/diagnostics.lua:466]: I see, phpstan will do nothing in this case, I remove the condition.

File: lua/null-ls/builtins/diagnostics.lua (hunk 1 above)
Comment (gbprod) [reply lua/null-ls/builtins/diagnostics.lua]: I've tried something to get the temp_path from params, I let you check this :)

**Comment by jose-elias-alvarez** [comment]
Thanks for making the changes! Could you try merging this with 8d2e103f6cba4e2dcb1d8cd4bfd45357699b84ce and see if `params.temp_path` works as expected?

**Comment by gbprod** [comment]
Works as expected :ok_hand: 
Thanks for your job !

**Comment by jose-elias-alvarez** [comment]
No problem, thanks again!

**Pull request merged**